tribits/ci_support/CDashQueryAnalyzeReport.py

```diff
@@ -2193,6 +2410,415 @@ def createCDashTestHtmlTableStr(
+  def reportTestsets(self, testsLOD):
+    testDictsByTestsetAcro = binTestDictsByTestsetAcro(testsLOD)
+    for testsetAcro in self.testsetAcroList:
+      testsetLOD = testDictsByTestsetAcro.get(testsetAcro, None)
+      if testsetLOD:
+        testsetTypeInfo = getStandardTestsetTypeInfo(testsetAcro)
+        self.singleTestsetReporter.reportSingleTestset(testsetTypeInfo,
+          len(testsetLOD), testsetLOD)
+  # ToDo: Modify the above to assert that there are no unexpected test-set
+  # types!
+
+
+  # Generate a pass/fail report for all of the testset analyzed by
+  # reportTestsets().
+  def getTestsHtmlReportStr(self, testsSummaryTitle, useDetailsBlock=False,
+      detailsBlockSummary=None,
+    ):
+    cdashReportData = self.cdashReportData
+    cdashReportData.htmlEmailBodyTop = \
+      "<p>\n" + cdashReportData.htmlEmailBodyTop + "</p>\n"
+    return getFullCDashHtmlReportPageStr(
+      cdashReportData, pageTitle=testsSummaryTitle, pageStyle="",
+      detailsBlockSummary=detailsBlockSummary)
+
+
+# Bin a list of test dicts by issue tracker
+#
+# Input arguments:
+#
+#   testsLOD [in]: Tests list of dicts
+#
+# Returns (testDictsByIssueTracker, testsWithoutIssueTrackersLOD)
+#
+#  testDictsByIssueTracker [out]: A dict where the keys are the issue tracker
+#  strings (e.g. '#1234') and the values are the sublist test dicts that have
+#  that issue tracker.
+#
+#  testsWithoutIssueTrackersLOD [out]: The remaining list of test dicts that
+#  don't have an issue tracker.
+#
+# For example, the input list of dicts:
+#
+#  testsLOD = [
+#    { 'testname':'test1' ... 'issue_tracker':'#1234' },
+#    { 'testname':'test2' ... },
+#    { 'testname':'test3' ... 'issue_tracker':'#1235' },
+#    { 'testname':'test4' ... 'issue_tracker':'#1234' },
+#    { 'testname':'test5' ... },
+#    { 'testname':'test6' ... 'issue_tracker':'#1235' },
+#    { 'testname':'test7' ... 'issue_tracker':'#1236' },
+#
+# would yeild:
+#
+#   testDictsByIssueTracker = {
+#     '#1234' : [
+#       { 'testname':'test1' ... 'issue_tracker':'#1234' },
+#       { 'testname':'test4' ... 'issue_tracker':'#1234' },
+#       ],
+#     '#1235' : [
+#       { 'testname':'test3' ... 'issue_tracker':'#1235' },
+#       { 'testname':'test6' ... 'issue_tracker':'#1235' },
+#       ],
+#     '#1236' : [
+#       { 'testname':'test7' ... 'issue_tracker':'#1236' },
+#       ],
+#     ]
+#
+#   testsWithoutIssueTrackersLOD = [
+#    { 'testname':'test2' ... },
+#    { 'testname':'test5' ... },
+#    ]
+#
+def binTestDictsByIssueTracker(testsLOD):
+  testDictsByIssueTracker = {}
+  testsWithoutIssueTrackersLOD = []
+  for testDict in testsLOD:
+    issueTracker = testDict.get('issue_tracker', None)
+    if issueTracker:
+      issueTrackerBinTestsLOD = testDictsByIssueTracker.setdefault(issueTracker, [])
+      issueTrackerBinTestsLOD.append(testDict)
+    else:
+      testsWithoutIssueTrackersLOD.append(testDict)
+  return (testDictsByIssueTracker, testsWithoutIssueTrackersLOD)
```

Comment:
Make sure this is tested for the list `testsWithoutIssueTrackersLOD` non-empty.

Reply:
Check.  It is tested in the `test_binTestDictsByIssueTracker` unit tests `test_issues_1_noissues_1` and `test_issues_3_noissues_2`.